File: src/etch/bindings/csharp/compiler/test/TestTest1Helper.cs
using System;
using System.Collections.Generic;
using System.Text;
using NUnit.Framework;
using Etch.Transport;
using Etch.Util;
using etch.bindings.csharp.compiler.test;

namespace Etch.Compiler.Test
{
    [TestFixture]
    public class TestTest1Helper
    {
        public static Etch.Transport.Transport<SessionListener> listener;
        
        [TestFixtureSetUp]
        public void First()
        {
            Console.WriteLine();
            Console.Write( "TestHelper" );
        }

        [SetUp]
        public void Method_OpenNewTestListener()
        {
            TransportFactory.Add( "blah", new TcpTransportFactory() );

            string uri = "blah://localhost:4040";

            MainTest1Listener implFactory = new MainTest1Listener();

            listener = Test1Helper.NewListener( uri, null, implFactory );
            listener.TransportControl( TransportConsts.START_AND_WAIT_UP, 4000 );
        }

        [Test]
        public void Method_NewTestServer()
        {
            Boolean flag = true;

            string uri = "blah://localhost:4040";

            MainTest1Client client = new MainTest1Client();

            RemoteTest1Server server;

            try
            {
                server = Test1Helper.NewServer( uri, null, client );
                server._StartAndWaitUp( 4000 );

                // Insert your code here
                // -----------------------------------------------------------------------

                server._Stop();
            }
            catch ( Exception e )
            {
                flag = false;
                Console.WriteLine( e.StackTrace );
                Assert.IsTrue(flag);
            }

            Assert.IsTrue( flag );
        }

        [TearDown]
        public void Method_CloseNewTestListener()
        {
            listener.TransportControl( TransportConsts.STOP_AND_WAIT_DOWN, 0 );
        }
    }
}

File: src/etch/bindings/java/compiler/test/TestTest1Helper.java
package etch.bindings.java.compiler.test;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import etch.bindings.java.transport.TcpTransportFactory;
import etch.bindings.java.transport.TransportFactory;
import etch.util.core.io.Transport;

/** Test of Test1Helper */
public class TestTest1Helper
{
	private static Transport<?> listener;
	
	/** @throws Exception */
	@BeforeClass
	public static void init() throws Exception
	{
		TransportFactory.put( "rohit", new TcpTransportFactory() );
		
		String uri = "rohit://0.0.0.0:4040";
		
		listener = Test1Helper.newListener( uri, null, new MainTest1Listener() );
		
		listener.transportControl( Transport.START_AND_WAIT_UP, 4000 );
	}
	
	/** @throws Exception */
	@Test
	public void test() throws Exception
	{
		String uri = "rohit://localhost:4040";

		RemoteTest1Server server = Test1Helper.newServer( uri, null, new MainTest1Client() );
		server._startAndWaitUp( 4000 );

		// TODO nothing to do but wait for server to really be up.
		Thread.sleep( 1000 );
	
		server._stopAndWaitDown( 4000 );
		server = null;
	}
	
	/** @throws Exception */
	@AfterClass
	public static void fini() throws Exception
	{
		listener.transportControl( Transport.STOP_AND_WAIT_DOWN, 0 );
		listener = null;
	}
}
